File: solutions/kotlin/gigasecond/1/src/main/kotlin/Gigasecond.kt
import java.time.LocalDateTime
import java.time.LocalDate

class Gigasecond(ldt: LocalDateTime) {
    constructor(lt: LocalDate) : this(lt.atStartOfDay())

    val date: LocalDateTime = ldt.plusSeconds(Math.pow(10.0, 9.0).toLong())
}

File: solutions/scala/gigasecond/1/src/main/scala/Gigasecond.scala
import java.time.LocalDate
import java.time.LocalDateTime

object Gigasecond {
  val GIGASECOND = 1000000000

  def add(startDate: LocalDate): LocalDateTime = startDate.atStartOfDay().plusSeconds(GIGASECOND)

  def add(startDateTime: LocalDateTime): LocalDateTime = startDateTime.plusSeconds(GIGASECOND)
}
